sal-service/src/main/java/org/ow2/proactive/sal/service/service/CloudService.java

```diff
@@ -239,17 +243,25 @@ public Boolean removeClouds(String sessionId, List<String> cloudIds, Boolean pre
                              cloud.getCloudType());
             }
             if (Boolean.TRUE.equals(tempFlag)) {
+                List<Deployment> toBeRemovedDeployments = new ArrayList<>();
                 if (cloud.getDeployments() != null) {
-                    LOGGER.info("Cleaning deployments from related tasks " + cloud.getDeployments().toString());
-                    cloud.getDeployments().forEach(deployment -> deployment.getTask().removeDeployment(deployment));
+                    LOGGER.info("Cleaning all deployments related to cloud \"{}\"", cloud.getCloudId());
+                    toBeRemovedDeployments.addAll(cloud.getDeployments());
+                    for (Deployment deployment : toBeRemovedDeployments) {
+                        repositoryService.deleteDeployment(deployment);
+                    }
                 }
-                LOGGER.info("Cleaning deployments from the cloud entry");
-                cloud.clearDeployments();
                 LOGGER.info("Cleaning node candidates");
+                List<String> toBeRemovedCloud = Collections.singletonList(cloud.getCloudId());
```

Comment:
Same, you don't need a new list, you can remove it and give directly `Collections.singletonList(cloud.getCloudId())`

Reply:
Here I don't need a list but it's more efficient. 
Otherwise "Collections.singletonList(cloud.getCloudId());" will be called 4 times, so from a performance point of view it's better to define a list variable.